src/GovUk.Education.ExploreEducationStatistics.Public.Data.Processor.Tests/Functions/ProcessCompletionOfNextDataSetVersionImportFunctionTests.cs

```diff
@@ -117,6 +124,1547 @@ private static Mock<TaskOrchestrationContext> DefaultMockOrchestrationContext(Gu
         }
     }
 
+    public abstract class CreateChangesTests(
+        ProcessorFunctionsIntegrationTestFixture fixture)
+        : ProcessCompletionOfNextDataSetVersionImportFunctionTests(fixture)
+    {
+        protected const DataSetVersionImportStage Stage = DataSetVersionImportStage.CreatingChanges;
+
+        protected async Task CreateChanges(Guid instanceId)
+        {
+            var function = GetRequiredService<ProcessCompletionOfNextDataSetVersionFunction>();
+            await function.CreateChanges(instanceId, CancellationToken.None);
+        }
+    }
+
+    public class CreateChangesTestsFilterTests(
+        ProcessorFunctionsIntegrationTestFixture fixture)
+        : CreateChangesTests(fixture)
+    {
+        [Fact]
+        public async Task Success()
```

Comment:
Right now, after trying to read some of this test code, I've moved into thinking the overhead and complexity is unnecessarily high and we should be doing better here.

Think if we refactor away the use of the mapping plan in the changelog generation, we can simplify everything drastically. We should be able to more easily write tests that look like the ones for indicators, geographic levels and time periods. Writing tests like the other metas is my strong preference as they're sooooo much easier to read and understand.

If we can do so in a reasonable amount of time, I'd strongly suggest moving in this direction.

Same applies for locations.

Reply:
If we agree the mappings don't need to be involved, then I agree the tests could be made much simpler! 

What are your opinions on the whole `ParentChange`/`OptionChange` stuff for the test set-up? I've written my thoughts on it before I left for my holidays, and I have also spoken to Ben about it. I wasn't satisfied with it and I believe it could be made better. It was the best I could come up with at the time to make the tests as simple as possible to read, and when I was in a rush and couldn't think straight. But there are 100% better ways of doing it and I'm not sure this way is super intuitive. It's **definitely** not how I would normally choose to write my tests, but for lengthy reasons I've tried to explain before, I ended up going down this route.

Reply:
the mapping complexity has now been removed. As has the `ParentChange`/`OptionChange` stuff.